### PR TITLE
WIP Additional cloud fixes

### LINKED ISSUE #240: After downloading a project the "Open project" button is not enabled
But if you refresh the project list, then it becomes enabled.
image

### PR DESCRIPTION
**Pull request by suricactus**
- [x] More info what is actually deleted when click on the delete project
- [x] Fix when the project was downloaded, then deleted, then downloaded again
- [x] Refresh projects list after a project is downloaded (fix #240)
- [x] Allow different owners, other than the current user
- [x] Show only your own projects, do not include public
- [x] Enforce permissions when updating a project
- [x] Fix alignment of the dynamic fields in "base layer" group box
- [x] Project extent - used in offline editing and base map export

### CODE REVIEW

File: qfieldsync/gui/project_configuration_widget.py [hunk 1]
```diff
@@ -62,9 +62,11 @@ def __init__(self, parent=None):
         self.areaOfInterestExtentWidget.setToolTip(
             self.tr("Leave null to use the current project zoom extent.")
         )
-        self.areaOfInterestExtentWidget.setMapCanvas(iface.mapCanvas())
         self.areaOfInterestExtentWidget.setNullValueAllowed(True)
 
+        if iface:
```
Comment (m-kuhn) [inline qfieldsync/gui/project_configuration_widget.py:67]: How can this be run without iface again?

Comment (suricactus) [reply qfieldsync/gui/project_configuration_widget.py:67]: I guess when run from commandline, iface is `None`, it failed the previous job

Comment (m-kuhn) [reply qfieldsync/gui/project_configuration_widget.py:67]: is the `gui` lib used from `cli`?

Comment (suricactus) [reply qfieldsync/gui/project_configuration_widget.py:67]: no, none of them is using the gui:
```
import os
import shutil
import tempfile

from qgis.core import Qgis, QgsOfflineEditing, QgsProject
from qgis.testing import start_app, unittest
from qgis.testing.mocked import get_iface

from qfieldsync.libqfieldsync import OfflineConverter
from qfieldsync.tests.utilities import test_data_folder

start_app()
```

**Pull request merged**